lib/archethic.ex

```diff
@@ -302,6 +308,29 @@ defmodule Archethic do
     end
   end
 
+  @doc """
+  Parse and execute the contract in the given transaction.
+  We assume the contract is parse-able.
+  """
+  @spec parse_and_execute_contract_at(
+          Contract.trigger_type(),
+          Transaction.t(),
+          nil | Transaction.t()
+        ) ::
+          {:ok, nil | Transaction.t()}
+          | {:error,
+             :invalid_triggers_execution
+             | :invalid_transaction_constraints
+             | :invalid_oracle_constraints
+             | :invalid_inherit_constraints}
+  def parse_and_execute_contract_at(
```

Comment:
Maybe we should expect the contract's instance as a parameter instead of a contract's transaction. Hence the parsing validation can be done upfront in the controller layer without blowing the process.